modules/std.lobster

```diff
@@ -106,8 +106,8 @@ def product(xs): return fold(xs, 1): _x * _y
 
 def zip(xs, ys): return map xs.length: [ xs[_], ys[_] ]
 
-def reverse(xs, fun): for(xs.length) i: fun(xs[xs.length - i - 1])
-def reverse_list(xs): return map(xs.length) i: xs[xs.length - i - 1]
+def reverse(xs, fun):        for(xs.length) i: fun(xs[xs.length - i - 1], xs.length - i - 1)
```

Comment:
Maybe nice to put `i = xs.length - i - 1` so to only compute that once? Lobster doesn't have a particularly advanced optimizer yet, so particularly for functions in `std` its worth to write the most optimal code manually :)

Also, there seem to be extra spaces in random places?

Reply:
Done; I for one prefer to dedupe my own sub-expressions, rather than posit the mythical "sufficiently advanced compiler"; I only hadn't done here for minimal change previously ;-)

Also introduced `reverse_map` on argument of "obviousness" and "completeness".

Walked back the, now obsolete, white space changes for vertical alignment (in the prior regime of compact one-liners).